garage/envs/dm_control/dm_control_env.py

```diff
@@ -14,49 +14,45 @@ class DmControlEnv(gym.Env, Serializable):
     Binding for [dm_control](https://arxiv.org/pdf/1801.00690.pdf)
     """
 
-    def __init__(
-            self,
-            domain_name,
-            task_name,
-            plot=False,
-            width=320,
-            height=240,
-    ):
+    def __init__(self, domain_name, task_name):
+        self._domain_name = domain_name
+        self._task_name = task_name
         self._env = suite.load(domain_name=domain_name, task_name=task_name)
-
-        self._total_reward = 0
-        self._render_kwargs = {'width': width, 'height': height}
-
-        if plot:
-            self._viewer = DmControlViewer()
-        else:
-            self._viewer = None
+        self._viewer = None
 
         # Always call Serializable constructor last
         Serializable.quick_init(self, locals())
 
     def step(self, action):
         time_step = self._env.step(action)
-        if time_step.reward:
-            self._total_reward += time_step.reward
         return Step(
-            flatten_observation(time_step.observation)["observations"],
+            flatten_observation(time_step.observation)['observations'],
             time_step.reward, time_step.step_type == StepType.LAST,
             **time_step.observation)
 
     def reset(self):
-        self._total_reward = 0
         time_step = self._env.reset()
-        return flatten_observation(time_step.observation)["observations"]
+        return flatten_observation(time_step.observation)['observations']
 
-    def render(self):
-        if self._viewer:
-            pixels_img = self._env.physics.render(**self._render_kwargs)
-            self._viewer.loop_once(pixels_img)
+    def render(self, mode='human'):
+        # pylint: disable=inconsistent-return-statements
+        if mode == 'human':
+            if not self._viewer:
+                title = 'dm_control {}.{}'.format(self._domain_name,
+                                                  self._task_name)
+                self._viewer = DmControlViewer(title=title)
+                self._viewer.launch(self._env)
+            self._viewer.render()
+            return None
+        elif mode == 'rgb_array':
+            return self._env.physics.render()
+        else:
+            raise NotImplementedError
 
     def close(self):
         if self._viewer:
-            self._viewer.finish()
+            self._viewer.close()
```

Comment:
Should we also delete the reference to `self._viewer` here?

Reply:
Theoretically Python should do that as soon as DmControlEnv goes out of scope.

Reply:
i will try it